lib/lrmd/lrmd_client.c

```diff
@@ -1311,6 +1312,32 @@ report_async_connection_result(lrmd_t * lrmd, int rc)
     }
 }
 
+static void
+tls_handshake_failed(lrmd_t *lrmd, int tls_rc, int rc)
+{
+    lrmd_private_t *native = lrmd->lrmd_private;
+
+    crm_warn("Disconnecting after TLS handshake with "
+             "Pacemaker Remote server %s:%d failed: %s",
+             native->server, native->port,
+             (rc == EPROTO)? gnutls_strerror(tls_rc) : pcmk_rc_str(rc));
+    gnutls_deinit(*native->remote->tls_session);
+    gnutls_free(native->remote->tls_session);
+    native->remote->tls_session = NULL;
+    lrmd_tls_connection_destroy(lrmd);
```

Comment:
For consistency it would be a good idea to report the result here

Reply:
I had wondered why that wasn't there in the first place, but I figured there had to be some reason.  Added.

Reply:
previously the callers did it, so those have to be removed